File: javascript/grade-school/grade-school.js
export class GradeSchool {

  constructor() {
    this._studentGrades = [];
  }

  add(name, grade) {
    this._studentGrades.push({name, grade});
  }

  roster() {
    const output = {};

    for (const item of this._studentGrades) {
      if (output[item.grade] == null) {
        output[item.grade] = [];
      }
    }

    for (const item of this._studentGrades) {
      output[item.grade].push(item.name);
    }

    for (const grade of Object.keys(output)) {
      output[grade] = output[grade].sort();
    }

    return output;
  }

  grade(value) {
    return this._studentGrades
      .filter(x => x.grade === value)
      .map(x => x.name)
      .sort();
  }

}

File: python/grade-school/grade_school.py
class School(object):
    def __init__(self):
        self.student_grades = []


    def add_student(self, name, grade):
        self.student_grades.append((name, grade))


    def roster(self):
        copied = self.student_grades[:]
        copied.sort(key=lambda x: f'{x[1]:03} {x[0]}')
        names = map(lambda x: x[0], copied)
        return list(names)


    def grade(self, grade):
        copied = self.student_grades[:]
        filtered = filter(lambda x: x[1] == grade, copied)
        mapped = map(lambda x: x[0], filtered)
        students = list(mapped)
        students.sort()
        return students
